File: backend/lib/L1_domain/entities/tracker/goal.py
#  Copyright (c) 2022. Alexandr Moroz
from .base import Importable, Statusable, TrackerEntity


class GoalStatus(TrackerEntity, Statusable):
    pass


class Goal(TrackerEntity, Importable):

    parent_id: int | None
    _parent: any

    status_id: int | None
    _status: GoalStatus | None

    @property
    def parent(self) -> any:
        return self._parent

    @property
    def status(self) -> GoalStatus:
        return self._status

File: frontend/lib/L2_data/mappers/goal.dart
// Copyright (c) 2022. Alexandr Moroz

import 'package:openapi/openapi.dart';

import '../../L1_domain/entities/goals/goal.dart';
import 'goal_status.dart';
import 'task.dart';

extension GoalMapper on GoalSchemaGet {
  Goal get goal => Goal(
        id: id,
        parentId: parentId,
        createdOn: createdOn.toLocal(),
        updatedOn: updatedOn.toLocal(),
        title: title,
        description: description ?? '',
        dueDate: dueDate?.toLocal(),
        closed: closed,
        status: status != null ? status!.status : null,
        tasks: tasks?.map((t) => t.task).toList() ?? [],
      );
}
